plugins/browser-plugin-ga-cookies/src/index.ts

```diff
@@ -1,54 +1,85 @@
+  ga4MeasurementId: string | string[];
+  cookiePrefix?: string;
+}
+
+const defaultPluginOptions: GACookiesPluginOptions = {
+  ua: true,
+  ga4: false,
+  ga4MeasurementId: '',
+  cookiePrefix: '',
+};
 
 /**
  * Captures the GA cookies on a page and sends as context on each event
+ * @param pluginOptions.ua - Send Universal Analytics specific cookie values.
+ * @param pluginOptions.ga4 - Send Google Analytics 4 specific cookie values.
+ * @param pluginOptions.ga4MeasurementId - Measurement id/ids to search the Google Analytics 4 session cookie. Can be a single measurement id as a string or an array of measurement id strings. The cookie has the form of _ga_<container-id> where <container-id> is the data stream container id.
+ * @param pluginOptions.cookiePrefix - Cookie prefix set on the Google Analytics 4 cookies.
  */
-export function GaCookiesPlugin(): BrowserPlugin {
+export function GaCookiesPlugin(pluginOptions: GACookiesPluginOptions = defaultPluginOptions): BrowserPlugin {
   return {
     contexts: () => {
-      const gaCookieData: SelfDescribingJson<Cookies> = {
-        schema: 'iglu:com.google.analytics/cookies/jsonschema/1-0-0',
-        data: {},
-      };
-      ['__utma', '__utmb', '__utmc', '__utmv', '__utmz', '_ga'].forEach(function (cookieType) {
-        var value = cookie(cookieType);
-        if (value) {
-          gaCookieData.data[cookieType] = value;
+      const contexts: SelfDescribingJson<Record<string, unknown>>[] = [];
+      const { ga4, ga4MeasurementId, ua, cookiePrefix } = { ...defaultPluginOptions, ...pluginOptions };
+      const GA_USER_COOKIE = '_ga';
+      const GA4_MEASUREMENT_ID_PREFIX = 'G-';
+      const GA4_COOKIE_PREFIX = '_ga_';
+
+      if (ua) {
+        const uaCookiesContext: SelfDescribingJson<UACookies> = {
+          schema: UNIVERSAL_ANALYTICS_COOKIES_SCHEMA,
+          data: {},
+        };
+        ['__utma', '__utmb', '__utmc', '__utmv', '__utmz', GA_USER_COOKIE].forEach(function (cookieType) {
+          var value = cookie(cookieType);
+          if (value) {
+            uaCookiesContext.data[cookieType] = value;
+          }
+        });
+        contexts.push(uaCookiesContext);
+      }
+
+      if (ga4) {
+        if (!ga4MeasurementId) {
+          LOG.warn('"ga4MeasurementId" option not provided in browser-plugin-ga-cookies.');
+          return contexts;
         }
-      });
-      return [gaCookieData];
+        const ga4CookiesContext: SelfDescribingJson<GA4Cookies> = {
+          schema: GOOGLE_ANALYTICS_4_COOKIES_SCHEMA,
+          data: {
+            _ga: '',
+            session_cookies: [],
+          },
+        };
+
+        const measurementIdentifiers = Array.isArray(ga4MeasurementId) ? [...ga4MeasurementId] : [ga4MeasurementId];
+
+        [...measurementIdentifiers, GA_USER_COOKIE].forEach(function (cookieIdentifier) {
+          if (cookieIdentifier === GA_USER_COOKIE) {
+            ga4CookiesContext.data._ga = cookie(cookiePrefix + cookieIdentifier);
```

Comment:
Nit: Just do this unconditionally outside of the loop.

Reply:
Fixed, good call.